### PR TITLE
new function app - deleteNode

### PR DESCRIPTION
**Pull request by lihi2910**
add deleteNode function app

### CODE REVIEW

File: CreateEvent/validation.ts [hunk 1]
```diff
@@ -14,4 +14,13 @@ const eventSchema = Joi.object({
...
 
+// eslint-disable-next-line import/first
+import * as moment from 'moment';
+
+const f = Joi.date();
+// new Date(moment('2021-08-02T09:30:00+10:00:00').format())
+// const result = f.validate(new Date(moment('2021-08-02T09:30:00+10:00:00').format())).error;
+// console.log('hiii', result);
+console.log(moment('2010-10-20T04:30:00.000+00:00').format());
+
```
Comment (maykremer) [inline CreateEvent/validation.ts:25]:
```suggestion
```

File: DeleteNode/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        const { id } = context.bindingData;
+
+        const nodeToDelete: Node = await NodeModel.findOne({ id });
+
+        // check if node not connected to NodeGroup
+        if (!nodeToDelete) {
+            throw new ObjectNotFoundFunctionError(functionName, 'node', id);
+        }
+
+        console.log(nodeToDelete.nodeGroupId);
```
Comment (maykremer) [inline DeleteNode/index.ts:31]:
```suggestion
```

File: DeleteNode/index.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        if (nodeToDelete.nodeGroupId) {
+            console.log('not possible');
```
Comment (maykremer) [inline DeleteNode/index.ts:33]:
```suggestion
```

File: DeleteNode/index.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            throw new ValidationFunctionError(
+                functionName,
+                `The node ${id} still have connected nodeGroup. Remove the nodeGroup before deleting the node`,
+            );
+        }
+        console.log(nodeToDelete);
```
Comment (maykremer) [inline DeleteNode/index.ts:39]:
```suggestion
```

File: DeleteNode/index.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        const deletedNode: Node = await NodeModel.findByIdAndDelete(nodeToDelete);
+        if (!deletedNode) {
+            console.log('not delete');
```
Comment (maykremer) [inline DeleteNode/index.ts:42]:
```suggestion
```

File: DeleteNode/index.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        console.log(nodeToDelete.nodeGroupId);
+        if (nodeToDelete.nodeGroupId) {
+            console.log('not possible');
+            throw new ValidationFunctionError(
+                functionName,
+                `The node ${id} still have connected nodeGroup. Remove the nodeGroup before deleting the node`,
+            );
+        }
+        console.log(nodeToDelete);
+        const deletedNode: Node = await NodeModel.findByIdAndDelete(nodeToDelete);
```
Comment (maykremer) [inline DeleteNode/index.ts:40]: You need to do `findOne` because the objectId is different than the `id` parameter

File: DeleteNode/index.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        if (!deletedNode) {
+            console.log('not delete');
+            throw new ObjectNotFoundFunctionError(functionName, 'node', id);
+        }
+
+        return {
+            body: deletedNode,
+        };
+    } catch (error) {
+        console.log(httpErrorHandler(error, functionName));
```
Comment (maykremer) [inline DeleteNode/index.ts:50]:
```suggestion
```

File: shared/node/node.model.ts [hunk 8]
```diff
@@ -22,7 +22,7 @@ const nodeSchema = new Schema(
         type: {
             type: String,
             enum: NodeType,
-            default: NodeType.Unknown,
+            // default: NodeType.Unknown,
```
Comment (maykremer) [inline shared/node/node.model.ts:25]: why?

**Review by maykremer — CHANGES_REQUESTED** [review]

File: DeleteNode/index.ts (hunk 6 above)
Comment (lihi2910) [reply DeleteNode/index.ts:40]: nodeToDelete include the objectId, I got it from the db

**Review by carmeltsadok — APPROVED** [review]

File: DeleteNode/index.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
+import { AzureFunction, Context, HttpRequest } from '@azure/functions';
+import { ForbiddenFunctionError, httpErrorHandler, ValidationFunctionError, ObjectNotFoundFunctionError } from '../shared/error/error';
+import getConnection from '../shared/mongoose';
+import { User, Role } from '../shared/user/user.interface';
+import authWrapper from '../shared/authWrapper';
+import HttpTriggerResult from '../shared/httpTriggerResult';
+import NodeGroup from '../shared/nodeGroup/nodeGroup.interface';
+import NodeGroupModel from '../shared/nodeGroup/nodeGroup.model';
```
Comment (carmeltsadok) [inline DeleteNode/index.ts:8]:
```suggestion
```

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

**Review by carmeltsadok — DISMISSED** [review]

File: DeleteNode/index.ts [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        const { id } = context.bindingData;
+
+        const nodeToDelete: Node = await NodeModel.findOne({ id });
+
+        // check if node not connected to NodeGroup
+        if (!nodeToDelete) {
+            throw new ObjectNotFoundFunctionError(functionName, 'node', id);
+        }
+
+        if (nodeToDelete.nodeGroupId) {
```
Comment (lustigy96) [inline DeleteNode/index.ts:29]:
```suggestion
        // check if node not connected to NodeGroup
        if (nodeToDelete.nodeGroupId) {
```

File: DeleteNode/index.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+            throw new ForbiddenFunctionError(functionName, user);
+        }
+
+        await getConnection();
+
+        const { id } = context.bindingData;
+
+        const nodeToDelete: Node = await NodeModel.findOne({ id });
+
+        // check if node not connected to NodeGroup
```
Comment (lustigy96) [inline DeleteNode/index.ts:24]:
```suggestion
```

File: DeleteNode/index.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+            throw new ValidationFunctionError(
+                functionName,
+                `The node ${id} still have connected nodeGroup. Remove the nodeGroup before deleting the node`,
+            );
+        }
+        const deletedNode: Node = await NodeModel.findByIdAndDelete(nodeToDelete);
+        if (!deletedNode) {
+            throw new ObjectNotFoundFunctionError(functionName, 'node', id);
+        }
+
```
Comment (lustigy96) [inline DeleteNode/index.ts:39]: choose between to tow conditions of the deletedNode

**Review by lustigy96 — CHANGES_REQUESTED** [review]

**Review by lustigy96 — APPROVED** [review]

**Pull request merged**